# Data exploration

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
salary_df = pd.read_excel('salary_data_states.xlsx')

read_excel() as this is an xlsx file, not csv.

In [3]:
type(salary_df)

pandas.core.frame.DataFrame

In [4]:
salary_df.shape

(167278, 27)

This tells us there are 167278 rows (specific jobs), and 27 columns (which are various bits of information associated with each job).

In [5]:
salary_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 167278 entries, 0 to 167277
Data columns (total 27 columns):
 #   Column                          Non-Null Count   Dtype  
---  ------                          --------------   -----  
 0   CASE_NUMBER                     167278 non-null  object 
 1   CASE_STATUS                     167278 non-null  object 
 2   CASE_RECEIVED_DATE              167278 non-null  object 
 3   DECISION_DATE                   167278 non-null  object 
 4   EMPLOYER_NAME                   167278 non-null  object 
 5   PREVAILING_WAGE_SUBMITTED       167278 non-null  float64
 6   PREVAILING_WAGE_SUBMITTED_UNIT  167278 non-null  object 
 7   PAID_WAGE_SUBMITTED             167278 non-null  float64
 8   PAID_WAGE_SUBMITTED_UNIT        167278 non-null  object 
 9   JOB_TITLE                       167278 non-null  object 
 10  WORK_CITY                       167275 non-null  object 
 11  EDUCATION_LEVEL_REQUIRED        11093 non-null   object 
 12  COLLEGE_MAJOR_RE

This gives us more information about the columns in this sheet - useful for getting more specific data later on.

In [6]:
salary_df.describe()

PREVAILING_WAGE_SUBMITTED  PAID_WAGE_SUBMITTED  \
count               1.672780e+05         1.672780e+05   
mean                7.115752e+04         8.164186e+04   
std                 3.874624e+04         4.147703e+04   
min                 5.050000e+00         5.050000e+00   
25%                 5.447500e+04         6.100000e+04   
50%                 6.970100e+04         7.700000e+04   
75%                 8.993700e+04         1.000000e+05   
max                 6.217100e+06         2.500000e+06   

       EXPERIENCE_REQUIRED_NUM_MONTHS  PREVAILING_WAGE_PER_YEAR  \
count                     4965.000000             167210.000000   
mean                        34.692044              74274.868236   
std                         22.317783              25356.245893   
min                          0.000000              10504.000000   
25%                         12.000000              56880.000000   
50%                         24.000000              70928.000000   
75%                         60.000000              90459.000000   
max                        144.000000             320000.000000   

       PAID_WAGE_PER_YEAR          order  
count        1.672780e+05  167278.000000  
mean         8.553277e+04   83714.716305  
std          3.873847e+04   48300.236431  
min          1.050000e+04       1.000000  
25%          6.300000e+04   41901.250000  
50%          7.860000e+04   83722.500000  
75%          1.000060e+05  125541.750000  
max          2.500000e+06  167361.000000

This is useful to get a general idea of the values within each column. I will use this to get any idea what data to exclude later on.

In [7]:
salary_df['JOB_TITLE_SUBGROUP'].value_counts()

software engineer        99364
business analyst         27811
assistant professor      18866
teacher                  13912
data analyst              3840
attorney                  1488
data scientist            1227
management consultant      770
Name: JOB_TITLE_SUBGROUP, dtype: int64

This tells me the 'JOB_TITLE_SUBGROUP' has assistant professor, attorney, business analyst, data analyst, data scientist, management consultant, software engineer, teacher as job types. Later, we will exclude teachers and attorneys from the set.

In [8]:
salary_df['JOB_TITLE'].value_counts()

SOFTWARE ENGINEER                                     54478
BUSINESS ANALYST                                      19516
SENIOR SOFTWARE ENGINEER                              11714
ASSISTANT PROFESSOR                                   10134
Software Engineer                                      3743
                                                      ...  
ASSISTANT PROFESSOR, NEGOTIATIONS AND CONFLICT MAN        1
ASSISTANT PROFESSOR SECURITY STUDIES                      1
ASSISTANT PROFESSOR-SOCIAL WORK                           1
LONGTERM SUBSTITUTE TEACHER                               1
SR SOFTWARE ENGINEER-ANDROID FRAMEWORK                    1
Name: JOB_TITLE, Length: 12589, dtype: int64